### PR TITLE
Migration of the Groovy tests to the Kotlin language

### PR DESCRIPTION
**Pull request by hsz**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -8,9 +11,14 @@ plugins {
...
-    tasks.withType<GroovyCompile> {
-        sourceCompatibility = "1.7"
-        targetCompatibility = "1.7"
+    tasks {
+        withType<GroovyCompile> {
+            sourceCompatibility = "1.7"
+            targetCompatibility = "1.7"
+        }
+        withType<KotlinCompile> {
+            kotlinOptions.jvmTarget = "1.8"
```
Comment (zolotov) [inline build.gradle.kts:20]: 1.7 was chosen as it minimal working version for Gradle, change it only if it was changed there too

File: src/test/kotlin/org/jetbrains/intellij/DownloadIntelliJSpec.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package org.jetbrains.intellij
+
+import org.gradle.api.plugins.BasePlugin
+import java.io.File
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+
+class DownloadIntelliJSpec : IntelliJPluginSpecBase() {
```
Comment (zolotov) [inline src/test/kotlin/org/jetbrains/intellij/DownloadIntelliJSpec.kt:9]: what happened with Stepwise annotation?

**Review by zolotov — CHANGES_REQUESTED** [review]
I'd like not to loose history of files. It contains many explanations of different hacks and reasons. Could you try to preserve it?

File: src/test/kotlin/org/jetbrains/intellij/IntelliJPluginSpecBase.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            .withArguments(*tasks, "--stacktrace")
+
+    fun tasks(groupName: String): List<String> = build(ProjectInternal.TASKS_TASK).output.lines().run {
+        val start = indexOfFirst { it.equals("$groupName tasks", ignoreCase = true) } + 2
+        drop(start).takeWhile(String::isNotEmpty).map { it.substringBefore(' ') }
+    }
+
+
+    protected fun directory(path: String) = File(dir, path).apply { mkdirs() }
+//
```
Comment (zolotov) [inline src/test/kotlin/org/jetbrains/intellij/IntelliJPluginSpecBase.kt:108]: ?

File: build.gradle.kts (hunk 1 above)
Comment (hsz) [reply build.gradle.kts:20]: I get it, but Scala – in the end – will be completely removed anyway.

File: src/test/kotlin/org/jetbrains/intellij/IntelliJPluginSpecBase.kt (hunk 3 above)
Comment (hsz) [reply src/test/kotlin/org/jetbrains/intellij/IntelliJPluginSpecBase.kt:108]: Missing leftover.

**Pull request closed without merging**